backend/tests/integration/controllers/posts-controller.test.js

```diff
@@ -0,0 +1,155 @@
+import mongoose from 'mongoose';
+import request from 'supertest';
+import Post from '../../../models/post.js';
+import server from '../../../server.js';
+import { validCategories } from '../../../utils/constants.js';
+import { createPostObject } from '../../utils/helper-objects.js';
+
+afterAll(async () => {
+  await mongoose.disconnect();
+});
+
+let postId;
+const invalidPostId = '609c16c69405b14574c99999';
+describe('Integration Tests: Post creation', () => {
+  it('Post creation: Success - All fields are valid', async () => {
+    const response = await request(server).post('/api/posts').send(createPostObject());
+    expect(response.status).toBe(200);
+    expect(response.body).toHaveProperty('_id');
+    postId = response.body._id;
+
+    const fetchedPost = await Post.findById(postId);
```

Comment:
Isn't these expect unnecessary?
Your ensuring what I have created do exsist, 
which for me is unnecessary.